airbyte-cdk/bulk/core/load/src/main/kotlin/io/airbyte/cdk/command/DestinationStream.kt

```diff
@@ -15,6 +15,8 @@ import jakarta.inject.Singleton
  *
  * TODO: Add dedicated schema type, converted from json-schema.
  */
+// This looks very much like it should be a data class.
+// This way, you don't need to override hashCode and friends.
 class DestinationStream(val descriptor: Descriptor) {
```

Comment:
This class is incomplete. It will have schema and sync metadata like generation_id, with the descriptor as the natural key.

Arguably however, we should just use `DestinationStream.Descriptor` as a key for lookups by stream and not do any magic with the dashcode.